src/Controller/QuestionTemplateController.php

```diff
@@ -80,18 +82,24 @@ public function deleteQuestion(Request $request, array $requestAttributes): Resp
         return $response;
     }
 
-    public function getQuestions(Request $request, array $requestAttributes): Response
+    private function getCriteriaFromRequest(array $requestAttributes): Criteria
     {
         $filters = isset($requestAttributes['text']) ? ['text' => $requestAttributes['text']] : [];
-        $count = $this->questionTemplateService->getQuestionNumber($filters);
-        $paginator = new Paginator($count);
-        if (isset($requestAttributes['page'])) {
-            $paginator->setCurrentPage($requestAttributes['page']);
-        }
-        $questions = $this->questionTemplateService->getQuestions($filters, $paginator->getCurrentPage());
+        $currentPage = isset($requestAttributes['page']) ? $requestAttributes['page'] : 1;
```

Comment:
can be written as `$currentPage = $requestAttributes['page'] ?? 1;` :D just saying

Reply:
@Raresmldvn if $requestAttributes does not have value for 'page' wouldn't it result in a PHP error?

Reply:
no, just checked. it integrates with isset

Reply:
@Raresmldvn cool

Reply:
@andreiuta97 if you change this, merge after, if not, you can merge now if Victor also agrees

Reply:
@Raresmldvn, I'm gonna change it first.